update for run.javac testing wrt JDK-8341408

stephan-herrmann opened the pull request:
https://bugs.openjdk.org/browse/JDK-8341408 has been resolved 

Since preview features can be tested only against latest there is no use in keeping Excuse JavacBug8341408 => remove

stephan-herrmann commented:
Failure is
```
[2024-12-03T23:53:56.514Z] [ERROR] Failed to execute goal org.eclipse.tycho.extras:tycho-p2-extras-plugin:4.0.10:compare-version-with-baselines (compare-attached-artifacts-with-release) on project org.eclipse.jdt.core.tests.builder: Baseline and reactor have the same fully qualified version, but different content
[2024-12-03T23:53:56.514Z] [ERROR] different
[2024-12-03T23:53:56.514Z] [ERROR]    META-INF/ECLIPSE_.RSA: present in baseline only
[2024-12-03T23:53:56.514Z] [ERROR]    META-INF/ECLIPSE_.SF: present in baseline only
[2024-12-03T23:53:56.514Z] [ERROR]    resources/autoValueSnippet/libs/auto-value-1.6.5.jar: different
[2024-12-03T23:53:56.514Z] [ERROR]       META-INF/ECLIPSE_.RSA: present in baseline only
[2024-12-03T23:53:56.514Z] [ERROR]       META-INF/ECLIPSE_.SF: present in baseline only
[2024-12-03T23:53:56.514Z] [ERROR]    resources/autoValueSnippet/libs/auto-value-annotations-1.6.5.jar: different
[2024-12-03T23:53:56.514Z] [ERROR]       META-INF/ECLIPSE_.RSA: present in baseline only
[2024-12-03T23:53:56.514Z] [ERROR]       META-INF/ECLIPSE_.SF: present in baseline only
[2024-12-03T23:53:56.514Z] [ERROR]    workspace/Test571363.jar: different
[2024-12-03T23:53:56.514Z] [ERROR]       META-INF/ECLIPSE_.RSA: present in baseline only
[2024-12-03T23:53:56.514Z] [ERROR]       META-INF/ECLIPSE_.SF: present in baseline only
```

I have no idea, what difference is being detected here for files that haven't changed for months if not for years.

@iloveeclipse  Should we turn off these kinds of checks for BETA_JAVA24? They seem to cause more trouble than I can see benefit.

stephan-herrmann commented:
BTW, ignoring the problem is not a good idea as the build didn't even run all tests ...

iloveeclipse commented:
Looks like a tycho bug (or wrong config), @laeubi should know where these errors are coming from.

laeubi commented:
These files are usually ignored here

https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/blob/e673a526ac4a80f138c9642152ea9e3f9132c0e5/eclipse-platform-parent/pom.xml#L523-L528

so I can only guess that the config is somewhere overwritten, a version increment is missing, wrong baselinerepo is set or similar, @akurtakov and @mickaelistria  are probably more familiar about the setup for BETA branches.

jukzi commented:
I do not understand the title "testing wrt JDK-8341408".

stephan-herrmann commented:
> I do not understand the title "testing wrt JDK-8341408".

If you want to help, here's your crash course in compiler development in the face of difference between ecj and javac:
* run.javac is our testing mode that runs both compilers back to back with the same input and roughly the same expectation (generally without paying details to specific error messages from javac, just presence / absence of errors / warnings).
* [JDK-8341408](https://bugs.openjdk.org/browse/JDK-8341408) is a bug in javac 23
* Until that bug was fixed in javac 24, I implemented a specific `Excuse` that informs the test suite about the expected javac bug, so the test doesn't unnecessarily fail
* now that the bug is fixed and no version with that bug exists, that would be relevant for BETA_JAVA24, I removed the excuse, so now tests assert the actual equivalence of compilers regarding tests where previously they differed.
* The change is needed to get https://ci.eclipse.org/jdt/job/eclipse.jdt.core-run.javac-24 back on track.

HTH

jukzi commented:
> "testing wrt JDK-8341408".

What exactly does "wrt" mean or refer to? or is it some typo?

stephan-herrmann commented:
> > "testing wrt JDK-8341408".
> 
> What exactly does "wrt" mean or refer to? or is it some typo?

I'm more intested in getting a clean build than in discussing issue titles, but in case this is really important: wrt is my abbreviation of w.r.t. which is an abbreviation for with respect to. (de: "bzgl.")

stephan-herrmann commented:
To unblock development in BETA_JAVA24: can s.o. (= some one) please provide a hint how we can disable baseline comparison until the underlying problem is understood?

jukzi commented:
> (de: "bzgl.")

thanks for clarification

laeubi commented on Jenkinsfile:
```diff
@@ -30,7 +30,7 @@ pipeline {
+					mvn -U clean verify --batch-mode --fail-at-end -X -Dmaven.repo.local=$WORKSPACE/.m2/repository \
 						-Ptest-on-javase-24 -Pbree-libs -Papi-check -Pjavadoc -Pp2-repo \
 						-Dmaven.test.failure.ignore=true \
 						-Dcompare-version-with-baselines.skip=false \
```
```suggestion
						-Dcompare-version-with-baselines.skip=true \
```

You can try this if you just don't want to see the error (with possible other obstacles then), but this only hides the problem that the BETA_24 branch seems to be incorrectly configured (either with respect to the used baseline repo or git-parent, or ...) or your PR needs some changes.

For example if I look here it says the beta-branch is behind master (but that might be intended I don't know):

![grafik](https://github.com/user-attachments/assets/ba218455-70af-4184-ab95-45920d9d8e81)

If I look at `BETA_24`  then its last (merge) commit is **from last week**
![grafik](https://github.com/user-attachments/assets/58b4252f-f6d8-4b75-8343-e1e08c33617f)

and this PR contains commits from **2 week ago**

![grafik](https://github.com/user-attachments/assets/c83309bd-be10-4b1e-a4ba-2590270953be)

so maybe _this PR_ just needs to be rebased on top of `BETA_24` to fix the issue...

stephan-herrmann replied:
> You can try this if you just don't want to see the error ...

It seems skipping baseline checks is indeed our best option for now, simply because we have no indication what causes the check to check files which are marked to be ignored.

> For example if I look here it says the beta-branch is behind master (but that might be intended I don't know):

It's a long running branch, which means that is does not and cannot immediately reflect all changes in master.

> and this PR contains commits from 2 week ago

Right, but when those commits were freshly pushed 2 weeks ago they caused the same build failure. I think its the very nature of a commit from 2 weeks a ago that it was created 2 weeks ago, no?

> so maybe this PR just needs to be rebased on top of BETA_24 to fix the issue...

Just today I did exactly that rebase for #3390  and still after rebase we get the same bogus error. Ergo, we haven't understood the first thing about that error message.

laeubi replied:
> Just today I did exactly that rebase for https://github.com/eclipse-jdt/eclipse.jdt.core/pull/3390 and still after rebase we get the same bogus error. Ergo, we haven't understood the first thing about that error message.

I have already explained the error message before and according of you problem [here](https://github.com/eclipse-jdt/eclipse.jdt.core/pull/3446) it then looks like BETA_24 needs to be synced with master.

laeubi replied:
> sorry for being dense but I can't recall any statement from you that explained (in words which I can understand) why files META-INF/ECLIPSE_.RSA and META-INF/ECLIPSE_.SF should ever give a reason for build failure. Without any connection to the error message all the recommendations you may give are completely black box, voodoo. Nothing that will help me to improve the situation on my own, ever.

I elaborated the details on a recent example here:
- https://github.com/eclipse-jdt/eclipse.jdt.core/pull/3446#issuecomment-2547728849

Summary is given here:
- https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/2660#issuecomment-2547734985

if anything is still unclear please let me know.

Beside this its nothing voodo or only in my head but as platform builds are working for a long time (even before I started contributing to eclipse) so I can only take these "as-is" as other have decided over the details long before.
